typescript/infra/scripts/agent-utils.ts

```diff
@@ -122,39 +131,100 @@ export function assertEnvironment(env: string): DeployEnvironment {
   );
 }
 
-export function getEnvironmentConfig(environment: DeployEnvironment) {
-  return environments[environment];
-}
-
-export async function getConfigsBasedOnArgs(argv?: {
+// not requiring to build coreConfig to get agentConfig
+export async function getAgentConfigsBasedOnArgs(argv?: {
   environment: DeployEnvironment;
   context: Contexts;
+  'new-chains': string;
 }) {
-  const { environment, context = Contexts.Hyperlane } = argv
-    ? argv
-    : await withContext(getArgs()).argv;
-  const envConfig = getEnvironmentConfig(environment);
-  const agentConfig = getAgentConfig(context, envConfig);
-  return { envConfig, agentConfig, context, environment };
+  const {
+    environment,
+    context = Contexts.Hyperlane,
+    'new-chains': newChains,
+  } = argv ? argv : await withMissingChains(withContext(getArgs())).argv;
+
+  const newValidatorCounts: ChainMap<number> = {};
+  const newThresholds: ChainMap<number> = {};
+  if (newChains) {
+    const chains = newChains.split(',');
+    for (const chain of chains) {
+      const [chainName, threshold] = chain.split('=');
+      const [newThreshold, newValidatorCount] = threshold.split('/');
+      newThresholds[chainName] = Number(newThreshold);
+      newValidatorCounts[chainName] = Number(newValidatorCount);
+    }
+  }
+
+  const agentConfig = getAgentConfig(context, environment);
+  // check if new chains are needed
+  const missingChains = checkIfValidatorsArePresisted(agentConfig);
+
+  for (const chain of missingChains) {
+    if (!Object.keys(newValidatorCounts).includes(chain)) {
+      throw new Error(`Missing chain ${chain} not specified in new-chains`);
+    }
+    const baseConfig = {
+      [Contexts.Hyperlane]: [],
+      [Contexts.ReleaseCandidate]: [],
+      [Contexts.Neutron]: [],
+    };
+    // supplementing with dummy addresses for validator as part of missingChains
+    const validatorsConfig = validatorBaseConfigsFn(environment, context);
+
+    const validators = validatorsConfig(
+      {
+        ...baseConfig,
+        [context]: Array.from(
+          { length: newValidatorCounts[chain] },
+          () => '0x0',
+        ),
+      },
+      chain as Chains,
+    );
+    // the hardcoded fields are not strictly necessary to be accurate for create-keys.ts
+    // ideally would still get them from the chainMetadata
+    agentConfig.validators!.chains[chain] = {
+      interval: 1, // chainMetadata.estimateBlockNumber is optional
+      reorgPeriod: 0, // chainMetadata.reorgPeriod is optional
+      validators,
+    };
+  }
+
+  return {
+    agentConfig,
+    context,
+    environment,
+    newThresholds,
+  };
```

Comment:
why do we need to return all of this?

Reply:
We were already returning most of it except newThresholds which is needed for createAgentKeysIfNotExists